docs/user/managing-devices.md

```diff
@@ -0,0 +1,586 @@
+
+| Status Field | Description |
+| ------------ | ----------- |
+| | |
+
+### Managing Applications on the Web UI
+
+### Managing Applications on the CLI
+
+To deploy an application package from an OCI registry, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    image: quay.io/flightctl-demos/wordpress-app:latest
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+To deploy an unpackaged application from a Git repository, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    git:
+      url: https://github.com/flightctl/flightctl-demos.git
+      revision: main
+      path: /wordpress
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+To deploy an unpackaged application inline with the device specification, specify it in the device's `spec.applications[]` as follows:
+
+```yaml
+apiVersion: v1alpha1
+kind: Device
+metadata:
+  name: some_device_name
+spec:
+[...]
+  applications:
+  - name: wordpress
+    inline:
+      podman-compose.yaml: |
+        version: “3.7"
+        services:
+          wordpress:
+            image: “wordpress:latest”
+        [...]
+    envVars:
+      WORDPRESS_DB_HOST: "mysql"
+      WORDPRESS_DB_USER: "user"
+      WORDPRESS_DB_PASSWORD: "password"
+[...]
+```
+
+## Using Device Lifecycle Hooks
+
+You can specify device lifecycle hooks, which are custom actions the agent shall perform before or after certain lifecycle events. For example, you can add a shell script to your OS images that backs up your application data and then specify that this script shall be run and completed successfully before the agent can start updating the system.
+
+The following lifecycle hooks are supported:
+
+| Lifecycle Hook | Description |
+| -------------- | ----------- |
+| beforeUpdating | Specified actions are executed just before the agent starts updating the system. |
+| afterUpdating | Specified actions are executed after the agent has completed updating the system. |
+| beforeRebooting | Specified actions are executed just before the agent schedules a reboot. |
+| afterRebooting | Specified actions are executed when the agent restarts after a reboot. |
+
+An action can be to run an external command ("executable" action) or to update the state of systemd itself or a systemd unit ("systemd" action). When multiple actions are specified for a hook, these actions are performed in sequence, finishing one action before starting the next. **[TODO: Specify what happens when one or more actions fail]**
```

Comment:
Currently, when an action fails this is reported via status to the service. But a failure here does not fail the update.